Packs/ServiceNow/Integrations/ServiceNow_IAM/ServiceNow_IAM_test.py

```diff
@@ -0,0 +1,274 @@
+    """
+    client = mock_client()
+    args = {'user-profile': {'email': 'testdemisto2@paloaltonetworks.com'}}
+
+    mocker.patch.object(client, 'get_user', return_value=SERVICENOW_USER_OUTPUT)
+    mocker.patch.object(IAMUserProfile, 'update_with_app_data', return_value={})
+
+    user_profile = get_user_command(client, args, 'mapper_in')
+    outputs = get_outputs_from_user_profile(user_profile)
+
+    assert outputs.get('action') == IAMActions.GET_USER
+    assert outputs.get('success') is True
+    assert outputs.get('active') is True
+    assert outputs.get('id') == 'mock_id'
+    assert outputs.get('username') == 'mock_user_name'
+    assert outputs.get('details', {}).get('first_name') == 'mock_first_name'
+    assert outputs.get('details', {}).get('last_name') == 'mock_last_name'
+
+
+def test_get_user_command__non_existing_user(mocker):
+    """
+    Given:
+        - A ServiceNow IAM client object
+        - A user-profile argument that contains an email a user
+    When:
+        - The user does not exist in ServiceNow
+        - Calling function get_user_command
+    Then:
+        - Ensure the resulted User Profile object holds information about an unsuccessful result.
+    """
+    client = mock_client()
+    args = {'user-profile': {'email': 'testdemisto2@paloaltonetworks.com'}}
+
+    mocker.patch.object(client, 'get_user', return_value=None)
+
+    user_profile = get_user_command(client, args, 'mapper_in')
+    outputs = get_outputs_from_user_profile(user_profile)
+
+    assert outputs.get('action') == IAMActions.GET_USER
+    assert outputs.get('success') is False
+    assert outputs.get('errorCode') == IAMErrors.USER_DOES_NOT_EXIST[0]
+    assert outputs.get('errorMessage') == IAMErrors.USER_DOES_NOT_EXIST[1]
+
+
+def test_get_user_command__bad_response(mocker):
+    """
+    Given:
+        - A ServiceNow IAM client object
+        - A user-profile argument that contains an email of a non-existing user in ServiceNow
+    When:
+        - Calling function get_user_command
+        - A bad response (500) is returned from ServiceNow's API
+    Then:
+        - Ensure the resulted User Profile object holds information about the bad response.
+    """
+    import demistomock as demisto
+
+    client = mock_client()
+    args = {'user-profile': {'email': 'testdemisto2@paloaltonetworks.com'}}
+
+    bad_response = Response()
+    bad_response.status_code = 500
+    bad_response._content = b'{"error": {"detail": "details", "message": "message"}}'
+
+    mocker.patch.object(demisto, 'error')
+    mocker.patch.object(Session, 'request', return_value=bad_response)
+
+    user_profile = get_user_command(client, args, 'mapper_in')
+    outputs = get_outputs_from_user_profile(user_profile)
+
+    assert outputs.get('action') == IAMActions.GET_USER
+    assert outputs.get('success') is False
+    assert outputs.get('errorCode') == 500
+    assert outputs.get('errorMessage') == 'message: details'
```

Comment:
no need to get outputs from user profile.
instead validate that all the user_profile object contains the correct values, like:
```
assert user_profile.get_attribute('email') == 'testdemisto2@paloaltonetworks.com'
assert user_profile.get_action_results(0).action == False

Reply:
it means adding a `get_action_results` command to CSP (the action_results attribute is private), do we want that if we don't use it anywhere but in the test?